### PR TITLE
feat(cup): adds a labs section for cup

### PR DESCRIPTION
**Pull request by GeorgeMac**
This adds the labs section for `cup`.

It doesn't currently come with a step-by-step guide.
For now, it just includes all the moving parts (flipt, gitea, cup, grafana, whipt).

### CODE REVIEW

File: cup/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# Cup and Flipt
+
+<div align="center">
+  <img src="../images/cup.svg" alt="CUP" width="240" />
+</div>
+
+## Overview
+
+This application is designed to demonstrate [cup](https://github.com/flipt-io/cup), configured to manage Flipt flag state.
```
Comment (markphelps) [inline cup/README.md:9]: should we give a brief description of what cup is/does here too?

File: cup/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Cup and Flipt
+
```
Comment (markphelps) [inline cup/README.md:2]: want to add a link to this readme from the main readme too?

File: cup/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+- [Docker](https://docs.docker.com/get-docker/)
+- [Docker Compose](https://docs.docker.com/compose/install/)
+- [Cup CLI](https://github.com/flipt-io/cup) cloned and built from source
+
+## Usage
+
+1. `cd` into this directory (e.g. `cd cup`)
+1. Run `docker compose up --build`
+1. Run `cup defs` to connect to `cupd` and see available resources
```
Comment (markphelps) [inline cup/README.md:23]:
```suggestion
1. In another terminal window, run `cup defs` to connect to `cupd` and see available resources
```

File: cup/README.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+1. Run `cup get flags` to see currently served flags
```
Comment (markphelps) [inline cup/README.md:24]: is the thinking that we'll expand this to allow them to change the flag like you do in the demo?

Comment (GeorgeMac) [reply cup/README.md:24]: Yeah, I was thinking that would come when we add a full UI walkthrough like the other labs

**Review by markphelps — APPROVED** [review]
nice!

**Pull request merged**